robot/impl/local_robot_test.go

```diff
@@ -3134,19 +3182,468 @@ func TestMachineStatus(t *testing.T) {
+
+			remoteMd := cloud.Metadata{
+				PrimaryOrgID:  "the-remote-org",
+				LocationID:    "the-remote-location",
+				MachineID:     "the-remote-machine",
+				MachinePartID: "the-remote-part",
+			}
+			injectRemoteRobot.CloudMetadataFunc = func(context.Context) (cloud.Metadata, error) {
+				if !tc.remoteCloudMd {
+					return cloud.Metadata{}, errNoCloudMetadata
+				}
+				return remoteMd, nil
+			}
+			injectRemoteRobot.MachineStatusFunc = func(ctx context.Context) (robot.MachineStatus, error) {
+				// check that a timeout is passed down from the caller.
+				if _, ok := ctx.Deadline(); !ok {
+					return robot.MachineStatus{}, errors.New("no timeout detected")
+				}
+				md := cloud.Metadata{}
+				if tc.remoteCloudMd {
+					md = remoteMd
+				}
+				return robot.MachineStatus{
+					Resources: []resource.Status{
+						{
+							NodeStatus: resource.NodeStatus{
+								Name: resName1, State: resource.NodeStateUnconfigured, Revision: "rev2",
+							},
+							CloudMetadata: md,
+						},
+					},
+				}, nil
+			}
+			dRobot := newDummyRobot(t, injectRemoteRobot)
+			md := cloud.Metadata{
+				PrimaryOrgID:  "the-org",
+				LocationID:    "the-location",
+				MachineID:     "the-machine",
+				MachinePartID: "the-robot-part",
+			}
+			rev1 := "rev1"
+			cfg := &config.Config{}
+			if tc.localCloudMd {
+				cfg = &config.Config{
+					Cloud: &config.Cloud{
+						ID:           md.MachinePartID,
+						LocationID:   md.LocationID,
+						PrimaryOrgID: md.PrimaryOrgID,
+						MachineID:    md.MachineID,
+					},
+					Revision: rev1,
+				}
+			}
+			lr := setupLocalRobot(t, ctx, cfg, logger, withDisableBackgroundReconfiguration())
+			lr.(*localRobot).manager.addRemote(
+				context.Background(),
+				dRobot,
+				nil,
+				config.Remote{Name: remoteName},
+			)
+
+			mStatus, err := lr.MachineStatus(ctx)
+			test.That(t, err, test.ShouldBeNil)
+
+			expectedRev := ""
+			expectedMd := cloud.Metadata{}
+			if tc.localCloudMd {
+				expectedRev = rev1
+				expectedMd = md
+			}
+
+			expectedRemoteMd := cloud.Metadata{}
+			if tc.remoteCloudMd {
+				expectedRemoteMd = remoteMd
+			}
+			test.That(t, mStatus.Config.Revision, test.ShouldEqual, expectedRev)
+
+			expectedStatuses := rtestutils.ConcatResourceStatuses(
+				getExpectedDefaultStatuses(expectedRev, expectedMd),
+				[]resource.Status{
+					{
+						NodeStatus: resource.NodeStatus{
+							Name:  resource.NewName(client.RemoteAPI, remoteName),
+							State: resource.NodeStateReady,
+						},
+						CloudMetadata: expectedRemoteMd,
+					},
+					{
+						NodeStatus: resource.NodeStatus{
+							Name:  resName1.PrependRemote(remoteName),
+							State: resource.NodeStateReady,
+						},
+						CloudMetadata: expectedRemoteMd,
+					},
+				},
+			)
+			rtestutils.VerifySameResourceStatuses(t, mStatus.Resources, expectedStatuses)
+		})
+	}
+}
+
+func TestMachineStatusWithTwoRemotes(t *testing.T) {
+	// test that if one remote returns an error, MachineStatus returns correctly
```

Comment:
> MachineStatus returns correctly

[q] Sorry, what is `MachineStatus` supposed to return here? No error?

Reply:
no error and only cloud metadata for some resources